retry.go

```diff
@@ -109,20 +120,33 @@ func (r *Retry) postCheck(err error) bool {
 
 // Attempts sets the maximum amount of retry attempts
 // before the current error is returned.
-func (r *Retry) Attempts(n int) *Retry {
-	var iterations int
-	r.appendPreCondition(func() bool {
-		ok := iterations < n
-		iterations++
-		return ok
+// If maxAttempts is 0, then r.Run() will return a nil
+// error on any call.
+func (r *Retry) Attempts(maxAttempts int) *Retry {
+	i := 0
+	r.appendPreCondition(func(err error) error {
+		if maxAttempts == 0 {
```

Comment:
why doesn't this return an error?

Reply:
See https://github.com/codercom/retry/pull/17#discussion_r170979853

Though, that was in regards to whether to panic or not. I'm thinking setting Attempts to 0 might be an easy way to disable the retry. For example, if a command line tool exposes the number of attempts for a retry as a flag, the user could set it to 0 to disable the function the retry runs. Seems kinda janky though. 🤔 @nytopop what do you think?

Reply:
In the case of @nytopop agreeing. Why not just return the retry without modification if `maxAttempts` is 0?

Reply:
No reason, thats a great idea.

Reply:
Yeah, I'm good with returning the original. In that case, should `Attempts` change the number of times the function is *retried*, not the total number of times it will be run, to avoid the introduced inconsistency?

Reply:
We should change the behavior of `Attempts` to address that. `Attempts( <= 0 )` we should return nil once, and then return max attempts errors for subsequent calls. `Attempts(n > 0)` should call the function a maximum of `n + 1` times. Etc etc

Reply:
But then attempts would be be the number of times we retry the function instead of how many times we attempt to run it. I guess we could change the name. 🤔

Reply:
Lets rename it to `Reattempts` and deprecate `Attempts`?

Reply:
This does mean a consumer will never be able to stop the first run from happening. Not sure if it matters but we're losing a bit of flexibility here.

Reply:
So I say we leave the flexibility for now, never know when it'll be useful and its not a big deal otherwise.